c2corg_ui/static/js/viewdetails.js

```diff
@@ -168,6 +222,77 @@ app.ViewDetailsController.prototype.toggleTab = function(tab) {
 
 
 /**
+ * blablabla
+ * @export
+ */
+app.ViewDetailsController.prototype.initHeadband = function() {
+  if (this.documentService.document.activities.indexOf('rock_climbing') > -1 || this.documentService.document.activities.indexOf('mountain_climbing') > -1 || this.documentService.document.activities.indexOf('ice_climbing') > -1) {
+    this.hasHeadband = false;
+    if (this.documentService.document.associations.images.length == 0) {
+      this.hasVerticalImg = false;
+    } else {
+      this.hasVerticalImg = true;
+    }
+  } else {
+    this.hasVerticalImg = false;
+    if (this.documentService.document.associations.images.length == 0) {
+      this.hasHeadband = false;
+    } else if (this.documentService.document.associations.images.length == 1) {
+      this.hasHeadband = true;
+      this.scope_.headBands = this.createImageUrl(this.documentService.document.associations.images[this.widestImg_]['filename'],'BI');
+    } else if (this.documentService.document.associations.images.length > 1) {
+      this.getBestWideImg(0);
+      this.hasHeadband = true;
+    } else {
+      this.hasHeadband = false;
+    }
+  }
+};
+
+
+/**
+ * get the most wide image
+  * @param {number} index
+ * @return string
+ * @export
+ */
+app.ViewDetailsController.prototype.getBestWideImg = function(index) {
+  this.getMeta(index,this.createImageUrl(this.documentService.document.associations.images[index]['filename'],'MI'),function(index,w,h) {
+    if (this.widestCoef_ <= w / h) {
+      this.widestCoef_ = w / h;
+      this.widestImg_ = index;
+    }
+    if (index == this.documentService.document.associations.images.length - 1 || (this.widestCoef_ > 1 && this.documentService.document.associations.images.length >= 10 && index > 20)) {
+      this.scope_.headBands = this.createImageUrl(this.documentService.document.associations.images[this.widestImg_]['filename'],'BI');
+      this.scope_.$apply();
+    } else {
+      index++;
+      this.getBestWideImg(index);
+    }
+  }.bind(this));
+
+  return {'background-image': 'url(' + this.headBands + ')'};
+};
+
+
+/**
+ * load image
+ * @param {number} index
+ * @param {string} url
+ * @param {Function} callback
+ * @return string
+ * @export
+ */
+app.ViewDetailsController.prototype.getMeta = function(index,url, callback) {
```

Comment:
This name is not very explicit. What does "getMeta" mean?

Reply:
fixed